hikari/impl/shard.py

```diff
@@ -882,13 +880,21 @@ async def _run_once(self) -> bool:
 
         exit_stack = contextlib.AsyncExitStack()
 
+        query = {"v": _VERSION, "encoding": "json"}
+        if self._transport_compression:
+            query["compress"] = "zlib-stream"
+
+        url = self._resume_gateway_url or self._gateway_url
+        scheme, netloc, path, params, _, _ = urllib.parse.urlparse(url, allow_fragments=True)
+        encoded_query = urllib.parse.urlencode(query)
```

Comment:
It is possible for `resume_gateway_url` to contain a query, so we should append to the already existing ones instead of ignoring them